src/extensions/snippets/SnippetsView/SnippetsWidgetBox.tsx

```diff
@@ -0,0 +1,184 @@
+import { createCalculationPool, HitherContext, HitherInterface } from '../../common/hither';
+import useFetchCache from '../../common/useFetchCache';
+import { Recording, Sorting, SortingSelection, SortingSelectionDispatch } from '../../extensionInterface';
+import SnippetBox from './SnippetBox';
+
+
+type Props = {
+    recording: Recording
+    sorting: Sorting
+    noiseLevel: number
+    selection: SortingSelection
+    selectionDispatch: SortingSelectionDispatch
+    unitId: number
+    height: number
+}
+
+export type Snippet = {
+    timepoint: number
+    index: number
+    unitId: number
+    waveform?: number[][]
+}
+
+const calculationPool = createCalculationPool({maxSimultaneous: 6})
+
+interface SnippetsModelInterface {
+    getSnippets: (args: {timeRange?: {min: number, max: number} | null}) => Snippet[]
+}
+
+type InfoType = {
+    sampling_frequency: number
+    channel_ids: number[]
+    channel_locations: number[][]
+}
+
+type InfoQuery = {
+    type: 'info',
+    recording: Recording,
+    sorting: Sorting,
+    unitId: number
+}
+
+type SnippetsQuery = {
+    type: 'snippets',
+    recording: Recording,
+    sorting: Sorting,
+    unitId: number,
+    timeRange: {min: number, max: number}
+}
+
+type QueryType = InfoQuery | SnippetsQuery
+
+const getSnippetsInfo = async (args: {recording: Recording, sorting: Sorting, unitId: number, hither: HitherInterface}): Promise<InfoType> => {
+    const { recording, sorting, unitId, hither } = args
+    const result = await hither.createHitherJob(
+        'createjob_get_sorting_unit_info',
+        {
+            recording_object: recording.recordingObject,
+            sorting_object: sorting.sortingObject,
+            unit_id: unitId
+        },
+        {
+            useClientCache: true,
+            calculationPool
+        }
+    ).wait() as {
+        channel_ids: number[]
+        channel_locations: number[][]
+        sampling_frequency: number
+    }
+    return {
+        channel_ids: result.channel_ids,
+        channel_locations: result.channel_locations,
+        sampling_frequency: result.sampling_frequency
+    }
+}
+
+const getSnippets = async (args: {recording: Recording, sorting: Sorting, unitId: number, timeRange: {min: number, max: number}, hither: HitherInterface}): Promise<Snippet[]> => {
+    const { recording, sorting, unitId, timeRange, hither } = args
+    const result = await hither.createHitherJob(
+        'createjob_get_sorting_unit_snippets',
+        {
+            recording_object: recording.recordingObject,
+            sorting_object: sorting.sortingObject,
+            unit_id: unitId,
+            time_range: timeRange
+        },
+        {
+            useClientCache: true,
+            calculationPool
+        }
+    ).wait() as {
+        channel_ids: number[]
+        channel_locations: number[][]
+        sampling_frequency: number
+        snippets: Snippet[]
+    }
+    return result.snippets
+}
+
+const segmentSize = 30000 * 10
```

Comment:
Why this size?

Reply:
If we assume that sampling rate is 30 kHz and the firing rate of the unit is 10 Hz, then this would give us around 100 snippets. I was aiming for around 100 snippets retrieved per segment. No real reason.